src/WalletElementRenderer.res

```diff
@@ -0,0 +1,81 @@
+open PaymentType
+open RecoilAtoms
+
+@react.component
+let make = (~paymentType: CardThemeType.mode) => {
+  let sessionsObj = Recoil.useRecoilValueFromAtom(sessions)
+  let {paymentMethodOrder} = Recoil.useRecoilValueFromAtom(optionAtom)
+  let isApplePayReady = Recoil.useRecoilValueFromAtom(isApplePayReady)
+  let isGooglePayReady = Recoil.useRecoilValueFromAtom(isGooglePayReady)
+  let methodslist = Recoil.useRecoilValueFromAtom(list)
+  let paymentOrder = paymentMethodOrder->Utils.getOptionalArr->Utils.removeDuplicate
+  let (sessions, setSessions) = React.useState(_ => Dict.make()->JSON.Encode.object)
+  let (walletOptions, setWalletOptions) = React.useState(_ => [])
+
+  let (list, setList) = React.useState(_ => PaymentMethodsRecord.defaultList)
+
+  let areAllGooglePayRequiredFieldsPrefilled = DynamicFieldsUtils.useAreAllRequiredFieldsPrefilled(
+    ~list,
+    ~paymentMethod="wallet",
+    ~paymentMethodType="google_pay",
+  )
+
+  let areAllApplePayRequiredFieldsPrefilled = DynamicFieldsUtils.useAreAllRequiredFieldsPrefilled(
+    ~list,
+    ~paymentMethod="wallet",
+    ~paymentMethodType="apple_pay",
+  )
+
+  let walletList = React.useMemo(() => {
+    switch methodslist {
+    | Loaded(paymentlist) =>
+      let paymentOrder =
+        paymentOrder->Array.length > 0 ? paymentOrder : PaymentModeType.defaultOrder
+      let plist = paymentlist->Utils.getDictFromJson->PaymentMethodsRecord.itemToObjMapper
+      let (wallets, _) =
+        plist->PaymentUtils.paymentListLookupNew(
+          ~order=paymentOrder,
+          ~showApplePay=isApplePayReady,
+          ~showGooglePay=isGooglePayReady,
+          ~areAllGooglePayRequiredFieldsPrefilled,
+          ~areAllApplePayRequiredFieldsPrefilled,
+        )
+      wallets->Utils.removeDuplicate->Utils.getWalletPaymentMethod(paymentType)
+    | _ => []
+    }
+  }, (
+    methodslist,
+    paymentMethodOrder,
+    isApplePayReady,
+    isGooglePayReady,
+    areAllGooglePayRequiredFieldsPrefilled,
+    areAllApplePayRequiredFieldsPrefilled,
+  ))
+
+  React.useEffect(() => {
+    switch methodslist {
+    | Loaded(paymentlist) =>
+      let plist = paymentlist->Utils.getDictFromJson->PaymentMethodsRecord.itemToObjMapper
+
+      setWalletOptions(_ => walletList)
+      setList(_ => plist)
+    | _ => ()
+    }
+    None
+  }, (methodslist, walletList))
+  React.useEffect(() => {
+    switch sessionsObj {
+    | Loaded(ssn) => setSessions(_ => ssn)
+    | _ => ()
+    }
+    None
+  }, [sessionsObj])
```

Comment:
can you consume sessionsObj inside the PaymentRequestButton ?

Reply:
We can but PaymentRequestButton is being used inside PaymentElement as well, where we need to consume this sessionObj outside.